pkg/autodiscovery/listeners/docker.go

```diff
@@ -253,12 +200,28 @@ func (l *DockerListener) createService(ctx context.Context, cID string) {
 		log.Warnf("error while resolving image name: %s", err)
 		containerImage = ""
 	}
+
 	// Detect AD exclusion
 	containerName = cInspect.Name
 	if l.filters.IsExcluded(containers.GlobalFilter, containerName, containerImage, "") {
 		log.Debugf("container %s filtered out: name %q image %q", cID[:12], containerName, containerImage)
 		return
 	}
+
+	// Ignore containers that have been stopped for too long
+	if !cInspect.State.Running && cInspect.State.FinishedAt != "" {
+		finishedAt, err := time.Parse(time.RFC3339, cInspect.State.FinishedAt)
+		if err == nil {
+			excludeAfter := time.Duration(config.Datadog.GetInt("container_exclude_stopped_after")) * time.Hour
+			if finishedAt.Add(excludeAfter).Before(time.Now()) {
```

Comment:
This might be more naturally expressed as `time.Now().Sub(finishedAt) > excludeAge`?